Leaf class: bugfix around GetPMTType-macro

shotaizu opened the pull request:
I found a bug at a macro of "GetPMTType" in BQFitter.hh and fixed.
This causes miss-identification of the PMT type.

In addition,  I could not use the original BQFitter class in my code due to a linker-error.
I added "inline"  keyword and modified a declaration of "mtx"

Commit:
Original Makefile cannot make "lib" directory. Therefore I modified it.

 On branch LEAFClass
 Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

	modified:   .gitignore
	modified:   leaf/GNUmakefile

Commit:
Modified for use as library because I could not link them correctly.

 On branch LEAFClass
 Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

	modified:   leaf/BQFitter.cc
	modified:   leaf/BQFitter.hh

Commit:
I found a bug around GetPMTType macro and fixed

 On branch LEAFClass
 Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

	modified:   BQFitter.hh

gpronost commented:
Dear Izumiyama-san,

Regarding your modifications:
- Please remove the modification on N_THREAD (there is the function SetNThread to set the number of THREAD, you should use it instead).
- Define mtx as an extern may have some bad effect on code safety, if you want it to be defined in .cc remove it from the .hh (why do you need it in the .cc by the way? Is there some compilation trouble?)

gpronost commented:
Dear Izumiyama-san,

Thank you for your commits. Which compilator (and which version) did you use to compile?

I have some comments regarding your modifications however:
- Please remove the modification on N_THREAD (there is the function SetNThread to set the number of THREAD, you should use it instead).
- Define mtx as an extern may have some bad effect on code safety, if you want it to be defined in .cc remove it from the .hh (why do you need it in the .cc by the way? Is there some compilation trouble?)

shotaizu commented:
Dear gpronost-san,

Thank you for your review and comments.

I used GCC of veision "gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-11)" and succeeded compiling libBQFitter.so and the example-code ("analysis.cpp").

I understand about N_THREAD, I will use appropriate function.

However as for mtx, I caught an error at linking of my codes using class BQFitter.
I have two or more files (.cc), which include BQFitter.hh at each file in order to use "struct BQFitter::FitterOutput".
At linker process, "multiple definition of mtx" occurred and ld exited with status 1, and then I modified the mtx.

Commit:
On branch LEAFClass
 Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

	modified:   leaf/BQFitter.cc
	modified:   leaf/BQFitter.hh

Commit:
Restored N_THREAD and define WCSIM_single_PMT_type and BUG_WCGEO

 On branch LEAFClass
 Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

	modified:   leaf/BQFitter.cc
	modified:   leaf/BQFitter.hh

gpronost commented:
I see I understand, thank you for your modifications
